src/erc4626/erc4626.cairo

```diff
@@ -351,3 +368,10 @@ mod ERC4626Component {
         }
     }
 }
+
+impl ERC4626HooksEmptyImpl<TContractState> of ERC4626Component::ERC4626HooksTrait<TContractState> {
+    fn before_deposit(
+        ref self: ERC4626Component::ComponentState<TContractState>,
+        liquidity: u256,
```

Comment:
For standardisation, have all these params ` caller: ContractAddress,
            receiver: ContractAddress,
            assets: u256,
            shares: u256` in before_deposit